v2/client/client_default.go

```diff
@@ -12,7 +12,7 @@ import (
 // client, all outbound events will have a time and id set if not already
 // present.
 func NewDefault() (Client, error) {
-	p, err := http.New()
+	p, err := http.NewObserved()
```

Comment:
Can we avoid the default client to use the observed protocol? The user can easily create its own protocol and the related client

Reply:
Preserving trace context is the correct behavior for most users, I expect that there are very few use cases where dropping the incoming trace headers or ignoring the tracecontext in ctx is desirable. Using the NewObserved protocol will therefore create a more useful default behavior.
Note that observed is not really the best name for this protocol as trace propagation is orthogonal to exporting telemetry, I've chosen this name mostly for consistency with the observed client.

Reply:
My point is that we're introducing an hidden dependency and as @n3wscott pointed out https://github.com/cloudevents/sdk-go/pull/573#discussion_r478669817 that's something we wish to avoid. I see that this is the correct behaviour, but there might be users that are not using tracing, because the work to enable it is minimal (and already documented in examples etc) I prefer to avoid this change

Reply:
I'm not sure I follow. The opencensus libs are already used by client.NewObserved which is what the default client uses, so using the http.NewObserved protocol is not adding a new indirect dependency to the default client. There are no module level dependency changes as opencensus is already a dependency there as well.